docs/src/lib/binary_functions.md

```diff
@@ -64,26 +64,26 @@ monotone_chain!
 ## Intersection of two sets
 
 ```@docs
-intersection(::AbstractSingleton{N}, ::LazySet{N}) where {N<:Real}
-intersection(::Line2D{N}, ::Line2D{N}) where {N<:Real}
-intersection(::AbstractHyperrectangle{N}, ::AbstractHyperrectangle{N}) where {N<:Real}
-intersection(::Interval{N}, ::Interval{N}) where {N<:Real}
-intersection(::Interval{N}, ::HalfSpace{N}) where {N<:Real}
-intersection(::Interval{N}, ::Hyperplane{N}) where {N<:Real}
-intersection(::Interval{N}, ::LazySet{N}) where {N<:Real}
-intersection(::AbstractHPolygon{N}, ::AbstractHPolygon{N}, ::Bool=true) where {N<:Real}
-intersection(::AbstractPolyhedron{N}, ::AbstractPolyhedron{N}) where {N<:Real}
+intersection(::AbstractSingleton, ::LazySet)
+intersection(::Line2D, ::Line2D)
+intersection(::AbstractHyperrectangle, ::AbstractHyperrectangle)
+intersection(::Interval, ::Interval)
+intersection(::Interval, ::HalfSpace)
+intersection(::Interval, ::Hyperplane)
+intersection(::Interval, ::LazySet)
+intersection(::AbstractHPolygon, ::AbstractHPolygon, ::Bool=true)
+intersection(::AbstractPolyhedron{N}, ::AbstractPolyhedron{N}) where {N}
```

Comment:
the reason i kept `N` in this method is that it requires a  `_default_...(N)`. we can probably use the promotion there too.